homeassistant/helpers/entity.py

```diff
@@ -178,6 +179,23 @@ class DeviceInfo(TypedDict, total=False):
     default_model: str
 
 
+@dataclass
+class EntityDescription:
+    """An class that describes Home Assistant entities."""
+
+    # This is the key identifier for this entity
+    key: str
+
+    device_class: str | None = None
+    entity_picture: str | None = None
+    entity_registry_enabled_default: bool = True
+    force_update: bool = False
+    icon: str | None = None
+    name: str | None = None
+    supported_features: int | None = None
+    unit_of_measurement: str | None = None
```

Comment:
From what I've seen so far, I would like to suggest to only add those fields that are (highly) likely to be used.
By far the most common are `device_class`, `icon`, `name`, and `unit_of_measurement`.
One or two times I've also seen `entity_registry_enabled_default` (although named `enabled_by_default`).

IMO we could leave out `entity_picture`, `force_update`, and `supported_features` for now and add them if we notice that some integrations actually use define them in their sensor descriptions.

--
So far the changes where only related to `sensor` and `binary_sensor` platforms, so it might change once we look besides that.

Reply:
`entity_registry_enabled_default` matches the attribute name, hence the naming. I would object on changing it. I do not like it, because it is long, however, it is consistent.

`force_update` mainly occurs on energy-related sensors (like DSMR), I would like to keep it.

`entity_picture`, `supported_features` agree, lets ditch those.